### ISSUE TITLE
compatibility with python >= 3.7?

### ISSUE THREAD

**Issue by qihongl** [brainiak/brainiak#429]
It seems that the current brainiak doesn't work python version >= 3.7. Is there an easy fix?   

```
Qihong@:src$ conda install -c brainiak -c defaults -c conda-forge brainiak
Fetching package metadata .................
Solving package specifications: .

UnsatisfiableError: The following specifications were found to be in conflict:
  - brainiak -> python >=3.6,<3.7.0a0 -> readline 6.2*
  - brainiak -> python >=3.6,<3.7.0a0 -> sqlite 3.13.*
  - brainiak -> python >=3.6,<3.7.0a0 -> tk 8.5.*
  - python 3.7*
Use "conda info <package>" to see the dependencies for each package.
```

**Comment by mihaic** [comment]
Conda packages are specific to each Python minor version. We currently only have a package for Python 3.6. You can see what packages are available on the Anaconda website:
https://anaconda.org/brainiak/brainiak/files

As far as I know, there is nothing in BrainIAK incompatible with Python 3.7, but we have not yet built a Conda package for that version. However, I do have a vague recollection of encountering some issues with dependencies not being available for Python 3.7 when we released previous versions of BrainIAK. If a dependency is not available as a Conda package, we need to build it from PyPI and that requires customizing our Conda package build configuration. Currently, only `pymanopt` is installed from PyPI:
https://github.com/brainiak/brainiak/blob/master/.conda/meta.yaml

In conclusion, we should indeed build packages for more versions, but we should not increase the complexity of our build process too much as a consequence.

**Comment by mihaic** [comment]
Regarding the error message, I see instructions from Conda that suggest specifying the minor version of Python will make it work:
```
(py37)  mihai@exact:~$ conda install -c brainiak -c defaults -c conda-forge brainiak
Collecting package metadata (repodata.json): done
Solving environment: failed
Initial quick solve with frozen env failed.  Unfreezing env and trying again.
Solving environment: failed

UnsatisfiableError: The following specifications were found
to be incompatible with the existing python installation in your environment:

  - brainiak -> python[version='>=3.6,<3.7.0a0']

If python is on the left-most side of the chain, that's the version you've asked for.
When python appears to the right, that indicates that the thing on the left is somehow
not available for the python version you are constrained to.  Your current python version
is (python=3.7).  Note that conda will not change your python version to a different minor version
unless you explicitly specify that.
```

Indeed, specifying the Python version makes it work:
```
(py37) mihai@exact:~$ conda install -c brainiak -c defaults -c conda-forge brainiak python=3.6
Collecting package metadata (repodata.json): done
Solving environment: done
```

**Comment by cbaldassano** [comment]
Hi @mihaic, is there an ETA on when a 3.7 Conda package will be built? I know of several people here at Columbia that are currently choosing not to use Brainiak since they don't want to downgrade their python installation.

**Comment by mihaic** [comment]
Thanks for the feedback, @cbaldassano. We will make a bugfix release and include a Python 3.7 Conda package in the next few days.
